web/src/components/credentials/actions/CreateCredential.tsx

```diff
@@ -14,10 +14,49 @@ import {
   credentialTemplates,
   getDisplayNameForCredentialKey,
 } from "@/lib/connectors/credentials";
+import { getCurrentUser } from "@/lib/user";
+import { User, UserRole } from "@/lib/types";
 import { PlusCircleIcon } from "../../icons/icons";
 import { GmailMain } from "@/app/admin/connectors/[connector]/pages/gmail/GmailPage";
-import { ActionType, dictionaryType, formType } from "../types";
+import { ActionType, dictionaryType } from "../types";
 import { createValidationSchema } from "../lib";
+import { usePaidEnterpriseFeaturesEnabled } from "@/components/settings/usePaidEnterpriseFeaturesEnabled";
+import { AdvancedOptionsToggle } from "@/components/AdvancedOptionsToggle";
+import {
+  IsPublicGroupSelectorFormType,
+  IsPublicGroupSelector,
+} from "@/components/IsPublicGroupSelector";
+
+const CreateButton = ({
+  onClick,
+  isSubmitting,
+  isAdmin,
+  groups,
+}: {
+  onClick: () => void;
+  isSubmitting: boolean;
+  isAdmin: boolean;
+  groups: number[];
+}) => (
+  <div className="flex justify-end w-full">
+    <Button
+      className="enabled:cursor-pointer disabled:cursor-not-allowed disabled:bg-blue-200 bg-blue-400 flex gap-x-1 items-center text-white py-2.5 px-3.5 text-sm font-regular rounded-sm"
+      onClick={onClick}
+      type="button"
+      disabled={isSubmitting || (!isAdmin && groups.length === 0)}
+    >
+      <div className="flex items-center gap-x-1">
+        <PlusCircleIcon size={16} className="text-indigo-100" />
+        Create
+      </div>
+    </Button>
+  </div>
+);
+
+type formType = IsPublicGroupSelectorFormType & {
+  name: string;
+  [key: string]: any; // For additional credential fields
```

Comment:
`[key: string]: any` reduces type safety substantially - may be only way for this approach, but preferrably we don't let any additional keys with any values be set